domain/user/service/service.go

```diff
@@ -198,7 +223,19 @@ func (s *Service) AddUserWithActivationKey(ctx context.Context, user user.User)
 		return nil, fmt.Errorf("generating activation key for user %q: %w", user.Name, err)
 	}
 
-	if err = s.st.AddUserWithActivationKey(ctx, user, activationKey); err != nil {
+	// Generate a UUID for the user.
+	uuid, err := domainuser.NewUUID()
+	if err != nil {
+		return nil, fmt.Errorf("adding user %q with activation key, generating UUID: %w", user.Name, err)
+	}
+
+	// Get creator UUID for creatorUsername
+	creatorUser, err := s.st.GetUser(ctx, creatorUsername)
```

Comment:
Before generating the uuid we probably want to do this operation. We are also going to want to validate the creatorUsername through ValidateUsername for sanity checking before throwing unknown data at the state layer.

You could also call the service layer GetUser func instead to get all the goodness in one call.

Reply:
done